mint/git-clone/mint-leaf.yml

```diff
@@ -83,12 +86,49 @@ tasks:
 
       if [[ "${{ params.preserve-git-dir }}" == "false" ]]; then
         rm -rf .git
-      else
-        if [[ -n "$GITHUB_ACCESS_TOKEN" ]]; then
-          git config credential.helper '!bash -c "echo username=x-access-token && echo password=${GITHUB_ACCESS_TOKEN}"'
-        fi
       fi
     env:
       CACHE_BUST: ${{ tasks.get-latest-sha-for-ref.values.latest-sha-cache-buster }}
-      GIT_SSH_KEY: ${{ params.ssh-key }}
+      GIT_SSH_KEY:
+        value: ${{ params.ssh-key }}
+        cache: false
+      GITHUB_ACCESS_TOKEN:
+        value: ${{ params.github-access-token }}
+        cache: false
+
+  - key: jq
+    run: |
+      sudo apt-get update
+      sudo apt-get install jq
+      sudo apt-get clean
+  - key: configure-git
+    use: [git-clone, jq]
+    run: |
+      if [[ "${{ params.preserve-git-dir }}" == "false" ]]; then
+        exit 0
+      fi
+      if [[ -z "$GITHUB_ACCESS_TOKEN" ]]; then
+        exit 0
+      fi
+
+      git config credential.helper '!bash -c "echo username=x-access-token && echo password=${GITHUB_ACCESS_TOKEN}"'
+
+      QUERY="query { viewer { databaseId login } }"
+
+      ACCESS_TOKEN_DATA=$(curl \
+        --fail-with-body \
+        -H "Content-Type: application/json" \
+        -H "Authorization: Bearer ${GITHUB_ACCESS_TOKEN}" \
+        -X POST \
+        -d "{ \"query\": \"${QUERY}\"}" \
+        https://api.github.com/graphql \
+      )
+
+      GIT_USERNAME=$(echo $ACCESS_TOKEN_DATA | jq -r '.data.viewer.login')
+      ACCESS_TOKEN_ID=$(echo $ACCESS_TOKEN_DATA | jq -r '.data.viewer.databaseId')
+      GIT_EMAIL="${ACCESS_TOKEN_ID}+${GIT_USERNAME}@users.noreply.github.com"
+
+      git config user.email $GIT_EMAIL
+      git config user.name $GIT_USERNAME
+    env:
       GITHUB_ACCESS_TOKEN: ${{ params.github-access-token }}
```

Comment:
This one we do want to cache bust whenever the access token changes, so we can make sure to grab the right user/email for the access token.

Reply:
We have a functional test for this that you should be able to update to make sure `user.email` and `user.name` are set to the expected values.